### PR TITLE
Improving CLI 2.0 qa

### PR DESCRIPTION
**Pull request by byawitz**
## What does this PR do?

- Fix local development bug
- Consistent push messages 
- Improving wording
- Added ability to push functions without code
- Display a table of changes (when any) for all resources 
  - Including project settings

### CODE REVIEW

File: templates/cli/lib/questions.js.twig [hunk 1]
```diff
@@ -247,11 +247,10 @@ const questionsPullFunctions = [
             if (functions.length === 0) {
                 throw "We couldn't find any functions in your {{ spec.title|caseUcfirst }} project";
             }
-
             return functions.map(func => {
                 return {
                     name: `${func.name} (${func.$id})`,
-                    value: func
+                    value: { ...func }
```
Comment (Meldiron) [inline templates/cli/lib/questions.js.twig:253]: Is this change needed?

File: templates/cli/lib/config.js.twig [hunk 2]
```diff
@@ -8,7 +8,8 @@ const KeysFunction = new Set(["path", "$id", "execute", "name", "enabled", "logg
 const KeysDatabase = new Set(["$id", "name", "enabled"]);
 const KeysCollection = new Set(["$id", "$permissions", "databaseId", "name", "enabled", "documentSecurity", "attributes", "indexes"]);
 const KeysStorage = new Set(["$id", "$permissions", "fileSecurity", "name", "enabled", "maximumFileSize", "allowedFileExtensions", "compression", "encryption", "antivirus"]);
-const KeyTopics = new Set(["$id", "name", "subscribe"]);
+const KeysTopics = new Set(["$id", "name", "subscribe"]);
+const KeysTeams = new Set(["$id", "name"]);
 const KeyAttributes = new Set([
```
Comment (Meldiron) [inline templates/cli/lib/config.js.twig:13]: Can we also rename KeyAttributes to KeysAttributes? Seems only one without plural

File: templates/cli/lib/questions.js.twig (hunk 1 above)
Comment (engibeeer) [reply templates/cli/lib/questions.js.twig:253]: Yes,  won't work without. something to do with objects

File: templates/cli/lib/commands/push.js.twig [hunk 3]
```diff
@@ -658,7 +726,7 @@ const generateChangesObject = (attribute, collection, isAdding) => {
     return {
         key: `${chalk.yellow(attribute.key)} in ${collection.name} (${collection['$id']})`,
         attribute: attribute,
-        reason: isAdding ? 'Field doesn\'t exist on the remote server' : 'Field doesn\'t exist in appwrite.json file',
+        reason: isAdding ? 'Field isn\'t available on the remote server' : 'Field isn\'t available on the appwrite.json file',
```
Comment (Meldiron) [inline templates/cli/lib/commands/push.js.twig:729]:
```suggestion
        reason: isAdding ? 'Field isn\'t present on the remote server' : 'Field isn\'t present on the appwrite.json file',
```

Isn't available still sounds like warning a little.
Isnt present sounds like a cause.

This might be just me. If you are not onboard, we can ask Dennis to pick

File: templates/cli/lib/commands/command.js.twig [hunk 4]
```diff
@@ -86,11 +86,11 @@ const {{ service.name | caseLower }}{{ method.name | caseUcfirst }} = async ({
 {% endfor %}
 {% for method in service.methods %}
 {{service.name | caseLower }}
-    .command(`{{ method.name | caseCamel }}`)
+    .command(`{{ method.name | caseCamel | cli_kebab }}`)
```
Comment (Meldiron) [inline templates/cli/lib/commands/command.js.twig:89]:
```suggestion
    .command(`{{ method.name | cli_kebab }}`)
```
Should work too. Lets try.

**Review by Meldiron — APPROVED** [review]

**Review by eldadfux — APPROVED** [review]
